File: java/69.java
class Solution {
    public int mySqrt(int x) {
        if(x == 1){
            return 1;
        }
        long lo = 1;
        long hi = x;
        while(lo < hi){
            long mid = (lo+hi)/2;
            if(mid * mid == x){
                return (int)mid;
            }else if(mid*mid > x){
                hi = mid;
            }else{
                lo = mid+1;
            }
        }
        return (int)lo-1;
    }
}

File: go/69.go
func mySqrt(x int) int {
	if x == 0 {
		return 0
	}
	if x <= 3 {
		return 1
	}
	lo := 1
	hi := x
	for lo < hi {
		mid := (lo + hi) / 2
		if mid*mid == x {
			return mid
		}
		if mid*mid < x {
			lo = mid + 1
		} else {
			hi = mid
		}
	}
	return lo - 1
}